README.md

```diff
@@ -1,27 +1,196 @@
+The **broadcast-example-api** repo provides an example broadcast API to help demonstrate an implementation of a Subscribe button with [built-in consent](https://docs.xmtp.org/consent/subscribe) provided by XMTP.
+
+Follow these steps to set up an example broadcast API like the one backing the example subscribe-broadcast web page [deployed here](https://subscribe-broadcast.vercel.app/subscribe/button) and connected to the XMTP `dev` network.
+
+To get the most out of this repo, use it with the [subscribe-broadcast repo](https://github.com/xmtp/subscribe-broadcast). The subscribe-broadcast repo provides an example subscribe-broadcast web page that you can use with the example broadcast API to send broadcast messages and provide broadcast statuses. To learn how, see [Try the subscribe with built-in consent and broadcast flow](https://docs.xmtp.org/consent/subscribe-broadcast).
+
+**Note**: This project was created using `bun init` in bun v1.0.25. [Bun](https://bun.sh/) is a fast all-in-one JavaScript runtime.
+
+## Install dependencies
 
 ```bash
 bun install
 ```
 
-To run:
+## Get a broadcast key bundle
+
+For this example, you can use this script to get a key bundle for the wallet address you want to use to broadcast the message.
+
+First, pick the wallet address you want to use as the broadcast sending address. Ensure that it is enabled on the XMTP `dev` network.
+
+### Define a broadcast
+
+In `src/lib/boradcasterConfigs.ts`, remove all but the first object in the broadcast configuration array. When you are done, the array should look like this:
+
+```ts
+const broadcastConfigs: BroadcastConfig[] = [
+  {
+    address: "0x85583261a4c3ad6785Ac90BD8880393831F97F54",
+    greeting: "Welcome to Good morning XMTP Broadcasts",
+    id: "XMTP",
+  },
+];
+```
+
+Set the **address** key to the broadcast sending address.
+
+Set the **greeting** key to the welcome message you want to send when someone subscribes to the broadcast.
+
+For the purposes of this example, you can leave the **id** key as-is.
+
+### Install the XMTP SDK
+
+If not already installed, install the XMTP JavaScript SDK. You must be using Node.js >=20.
 
 ```bash
-yarn dev
+yarn add @xmtp/xmtp-js ethers
+```
+
+### Install TSX
+
+If not already installed, install TSX.
+
+```bash
+npm install -g tsx
+```
+
+### Create the script to generate the key bundle
+
+In the `broadcast-example-api` directory, create a `generateAndPrintKeys.ts` file with the following contents.
+
+Replace `<YOUR_PRIVATE_KEY>` with the private key for the wallet address you want to use as the broadcast address.
+
+```tsx
+import { Wallet } from "ethers";
+import { Client } from "@xmtp/xmtp-js";
+import type { Signer } from "ethers";
+
+const ENCODING = "base64"; // Use base64 encoding for a clean text output
+
+export const generateAndPrintKeys = async (signer: Signer) => {
+  const clientOptions = {
+    env: "production", // this works even if in .env, XMTP_ENV=dev. if set this to dev, key bundle doesn't work.
```

Comment:
I don't get what is going on with this. =)

In my .env file, XMTP_ENV=dev - and the key bundle this script generates when this value is set to production works.

However, if I set this value to dev in this script, to match the env in my .env file, the generated key bundle doesn't work.

Reply:
Hmm that doesn't seem right, they should have to match